### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by rodinux**
## Problem

- *Use v2 and upgrade to last 1.3.0 release*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by rodinux** [comment]
!testing

**Comment by ericgaspar** [comment]
!testme

**Comment by rodinux** [comment]
wait because it breaks with some extensions, we have a discussion about this problem...

File: doc/ADMIN.md
Comment (alexAubin) [inline doc/ADMIN.md:15]: Meh I don't know what to do with this ... this seems like useful info for garradin's users, but people already on Paheko are not going to be interested to have this info in their webadmin ...

File: scripts/install
Comment (alexAubin) [inline scripts/install:29]:
```suggestion
ynh_add_fpm_config
```

(assuming YunoHost 11.2, we can get rid of this)

File: scripts/upgrade
Comment (alexAubin) [inline scripts/upgrade:53]:
```suggestion
ynh_add_fpm_config
```

Same as for install step, assuming YunoHost 11.2

File: manifest.toml
Comment (alexAubin) [inline manifest.toml:20]:
```suggestion
yunohost = ">= 11.2"
```

cf proposed changes in install and upgrade for fpm config

**Review by alexAubin — APPROVED** [review]
LGTM apart from proposed improvements

File: doc/ADMIN.md
Comment (rodinux) [reply doc/ADMIN.md]: Ok, I have put these instructions on garradin, please add a review here https://github.com/YunoHost-Apps/garradin_ynh/pull/106
And remove from here

File: scripts/install
Comment (rodinux) [reply scripts/install]: Ok

File: scripts/upgrade
Comment (rodinux) [reply scripts/upgrade]: Ok

File: manifest.toml
Comment (rodinux) [reply manifest.toml]: ok

**Comment by rodinux** [comment]
Wait, we are not ready to push ! this release 1.3.0 have already some bugs, it is not the stable branch.
I have use the 1.3.0-RC2 but this release does not exist anymore ! now there is a 1.3.0-RC3 with another checksum... 
We should wait to a stable release before... the uploads seems broken tonight...

**Comment by rodinux** [comment]
!testme

**Comment by rodinux** [comment]
@alexAubin There's a lot of changes in this major release... Do you think is better advise to take a look on the Changelog: https://fossil.kd2.org/paheko/wiki?name=Changelog/1.3

**Comment by alexAubin** [comment]
Not sure why you are pinging me specifically, I have no expertise about this app in particular

But if you mean "we should encourage people to have a look at the changelog when upgrading to this specific version because there are major changes", then maybe yes (idk, you know better than me). To achive this you can write a specific notifcation for this upgrade, cf https://yunohost.org/fr/packaging_app_doc#pre-post-install-notes-pr :  

>create `doc/PRE_UPGRADE.d/{version}.md` to have a note displayed before upgrading to a version equal or higher than `{version}`

**Comment by rodinux** [comment]
> Not sure why you are pinging me specifically, I have no expertise about this app in particular
> 
> But if you mean "we should encourage people to have a look at the changelog when upgrading to this specific version because there are major changes", then maybe yes (idk, you know better than me). To achive this you can write a specific notifcation for this upgrade, cf https://yunohost.org/fr/packaging_app_doc#pre-post-install-notes-pr :
> 
> > create `doc/PRE_UPGRADE.d/{version}.md` to have a note displayed before upgrading to a version equal or higher than `{version}`

Thank you ! I am quiet now after have done few  tests, if an old plugin is no more compatible, it will not brake the app, just an adverstisment and deleting the pugin restore a clean environment.

**Comment by rodinux** [comment]
It works for me, just something I don't understand. When upgrading a folder seems duplicated... 

```
ls -ll /var/www/paheko__2/data/
total 14628
-rw-r--r-- 1 paheko__2 paheko__2 5652480 27 juil. 21:33 association.pre_upgrade-1.3.0-rc3.sqlite
-rw-rw---- 1 paheko__2 www-data  9310208 27 juil. 21:33 association.sqlite
drwxrwx--- 5 paheko__2 www-data     4096 27 juil. 21:33 cache
drwxrwx--- 4 paheko__2 www-data     4096 27 juil. 21:31 data
-rw-rw---- 1 paheko__2 www-data      189 27 juil. 21:33 index.html
drwxrwx--- 8 paheko__2 www-data     4096 27 juil. 21:33 plugins

ls -ll /var/www/paheko__2/data/data/
total 5536
-rw-rw---- 1 paheko__2 www-data 5652520 27 juil. 21:31 association.sqlite
drwxrwx--- 3 paheko__2 www-data    4096 27 juil. 21:31 cache
-rw-rw---- 1 paheko__2 www-data     189 23 mai   13:57 index.html
drwxrwx--- 2 paheko__2 www-data    4096 23 mai   13:57 plugins
```

**Comment by rodinux** [comment]
testme!

**Comment by rodinux** [comment]
!testme

**Comment by rodinux** [comment]
> It works for me, just something I don't understand. When upgrading a folder seems duplicated...
> 
> ```
> ls -ll /var/www/paheko__2/data/
> total 14628
> -rw-r--r-- 1 paheko__2 paheko__2 5652480 27 juil. 21:33 association.pre_upgrade-1.3.0-rc3.sqlite
> -rw-rw---- 1 paheko__2 www-data  9310208 27 juil. 21:33 association.sqlite
> drwxrwx--- 5 paheko__2 www-data     4096 27 juil. 21:33 cache
> drwxrwx--- 4 paheko__2 www-data     4096 27 juil. 21:31 data
> -rw-rw---- 1 paheko__2 www-data      189 27 juil. 21:33 index.html
> drwxrwx--- 8 paheko__2 www-data     4096 27 juil. 21:33 plugins
> 
> ls -ll /var/www/paheko__2/data/data/
> total 5536
> -rw-rw---- 1 paheko__2 www-data 5652520 27 juil. 21:31 association.sqlite
> drwxrwx--- 3 paheko__2 www-data    4096 27 juil. 21:31 cache
> -rw-rw---- 1 paheko__2 www-data     189 23 mai   13:57 index.html
> drwxrwx--- 2 paheko__2 www-data    4096 23 mai   13:57 plugins
> ```

Ok, I discover this mistake was here from a while, I have added a clean remove of the folder `data/data` and now this will erase this duplicate folder, but the release works fine. 

I am still waiting the stable release for this version, for now I test the 1.3.0-rc4, there is also a 1.3.0-rc5 coming soon. When the stable 1.3.0 (not release candidate) should be ready, I will merge this PR on master branch.

**Comment by rodinux** [comment]
!testme

**Comment by alexAubin** [comment]
(Soooo, what's holding us before merging ?)

File: doc/ADMIN.md
Comment (alexAubin) [inline doc/ADMIN.md:3]:
```suggestion
If you wish, you can add personal costum configurations by editing the `__INSTALL_DIR__/config.local.user.php` file. To see the possibilities, you can read the `__INSTALL_DIR__/config.dist.php` file first.
```

File: doc/ADMIN_fr.md
Comment (alexAubin) [inline doc/ADMIN_fr.md:4]:
```suggestion
Vous pouvez si vous le souhaitez ajouter des configurations personnelles en éditant le fichier `__INSTALL_DIR__/config.local.user.php`.
Pour voir les possibilités vous pouvez lire auparavant le fichier `__INSTALL_DIR__/config.dist.php`.
```

**Comment by rodinux** [comment]
> (Soooo, what's holding us before merging ?)

I wait until this branch is for now the release candidate for next major upgrade 1.3.0, but not yet stable...

Nevertheless It would be nice to first merge this PR #27 which is the upgrade from 1.2.9 to 1.2.11 (stable branch)

**Comment by rodinux** [comment]
!testme

**Comment by rodinux** [comment]
!testme

File: doc/PRE_UPGRADE.d/1.3.0.md
Comment (alexAubin) [inline doc/PRE_UPGRADE.d/1.3.0.md:1]:
```suggestion
We encourage you to have a look at the Changelog: https://fossil.kd2.org/paheko/wiki?name=Changelog/1.3 when upgrading to this version because there are major changes. If you have installed some plugins, not all will be compatibles.
```

**Review by alexAubin — APPROVED** [review]

**Comment by alexAubin** [comment]
... I don't get why there are a gazillion conflicts with master ... Many commits / branches were merged into master since the beginning of this branch, despite the fact that this is testing, and the recommended practice is that everything should go into testing before being merged in master ... Otherwise everything becomes a huge mess ...

File: scripts/upgrade
Comment (alexAubin) [inline scripts/upgrade:38]: That part aint great from a security point of view, this could accidentally allow access to any user of the machine if permissions are not set properly ... Is this really needed considering the `data` folder is listed in `--keep` in ynh_setup_source later ?

File: scripts/upgrade
Comment (alexAubin) [inline scripts/upgrade:91]: (cf previous comment)

File: scripts/upgrade
Comment (rodinux) [reply scripts/upgrade:38]: Ho, I explain you my problems... 
I have try with the line as before
 `ynh_setup_source --dest_dir="$install_dir" --keep="association.sqlite data config.local.user.php"`,  
but like this there was a strange folder duplicated `data/data` and some files no more needed. The developper of the  app told me something was wrong in the process of upgrade.  Normally the correct way for upgrading is something like (in french):

> pour éviter que des anciens fichiers traînent le mieux est de :
> - décompresser la nouvelle archive dans un répertoire vierge
> - y recopier le répertoire data uniquement
> - vérifier que tout marche bien
> - pointer le virtualhost sur ce nouveau répertoire (ou renommer le
>   nouveau répertoire à la place de l'ancien, peu importe).

So I have next added this suggestion from someone in the tchat:
`ynh_setup_source --dest_dir="$install_dir" --keep="association.sqlite data config.local.user.php" --full_replace=1` ,
but doing like will not keep the file for DB, association.sqlite. It is better beacause it stays like a fresh install and remove old folders no more needed.

Finally I decide try stay with this line, but backup the folder `data` before to copy it again as suggested by the developper...

Perhaps there is another way to do this ?

Comment (rodinux) [reply scripts/upgrade:38]: > That part aint great from a security point of view, this could accidentally allow access to any user of the machine if permissions are not set properly ... Is this really needed considering the `data` folder is listed in `--keep` in ynh_setup_source later ?

If I stay with this process  with `cp` and later `rsync`, the folder `data`  is not needed in the `--keep`

Comment (rodinux) [reply scripts/upgrade:38]: You told me if I do so it is better comment or remove
`ynh_system_user_create --username=$app --home_dir="$final_path"` ??
Is this possible to do ??

**Comment by rodinux** [comment]
!testme

**Comment by rodinux** [comment]
!testme

**Comment by rodinux** [comment]
> Why do I need `--force` ??

Perhaps something wrong with the cookies ?? in the folder `tmp` there is a folder `ynh-paheko-cookie.txt`

```
cat /tmp/ynh-paheko-cookie.txt 
# Netscape HTTP Cookie File
# https://curl.se/docs/http-cookies.html
# This file was generated by libcurl! Edit at your own risk.

#HttpOnly_paheko.yunohost.packaging	FALSE	/	TRUE	0	__c	bf2f32dff872378ff2c0
```

**Comment by ericgaspar** [comment]
!testme

File: doc/ADMIN.md
Comment (alexAubin) [inline doc/ADMIN.md:3]:
```suggestion
If you wish, you can add custom configurations inside `__INSTALL_DIR__/config.local.user.php`. To see the possibilities, you can read the `__INSTALL_DIR__/config.dist.php` file first.
```

**Review by alexAubin — COMMENTED** [review]
(So the help message for `install.init_main_permission` is already in master but I would still make more sense inside `doc/PRE_INSTALL.md` and `doc/PRE_INSTALL_fr.md` imho)

**Review by alexAubin — APPROVED** [review]
Cheers Eric :heart:

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
Why do we need all those config files?
`config.local.php`
`config.local.yunohost.php`
`config.local.user.php`

**Comment by ericgaspar** [comment]
Also I have open #35 with:
- only one config file
- I think it is interesting to have a proper data directory __DATA_DIR__ (so we don't need to save the sqlite database)
- cleaning permissions

File: scripts/upgrade
Comment (ericgaspar) [reply scripts/upgrade]: Why not use `data_dir` (`/home/yunohost.app/paheko`) to store all data so it is not affected during upgrade?

**Pull request merged**
